File: packages/java/src/main/java/com/tesote/sdk/v1/StatusClient.java
package com.tesote.sdk.v1;

import com.fasterxml.jackson.databind.JsonNode;
import com.tesote.sdk.Transport;
import com.tesote.sdk.internal.Json;
import com.tesote.sdk.models.Status;
import com.tesote.sdk.models.Whoami;

/**
 * v1 status + whoami. {@code status()} works without an API key.
 */
public final class StatusClient {
    private final Transport transport;

    public StatusClient(Transport transport) { this.transport = transport; }

    /** Public health check. Auth not required server-side. */
    public Status status() {
        JsonNode node = transport.request(Transport.Options.get("/status"));
        return Json.treeToValue(node, Status.class);
    }

    /** Identity of the API key owner. Requires auth. */
    public Whoami whoami() {
        JsonNode node = transport.request(Transport.Options.get("/whoami"));
        return Json.treeToValue(node, Whoami.class);
    }
}

File: packages/csharp/src/V1/StatusClient.cs
using System.Threading;
using System.Threading.Tasks;
using Tesote.Sdk.Internal;
using Tesote.Sdk.Models;

namespace Tesote.Sdk.V1;

/// <summary>v1 status + whoami endpoints (cross-version: <c>/status</c> and <c>/whoami</c>).</summary>
public sealed class StatusClient
{
    private readonly Transport _transport;

    internal StatusClient(Transport transport)
    {
        _transport = transport;
    }

    /// <summary>Public status check. Auth not required server-side, but the SDK sends the bearer anyway.</summary>
    public async Task<StatusResponse> GetAsync(CancellationToken ct = default)
    {
        var node = await _transport.RequestAsync(RequestOptions.Get("/status"), ct).ConfigureAwait(false);
        return Json.Deserialize<StatusResponse>(node);
    }

    /// <summary>Identifies the bearer-token holder (workspace or user).</summary>
    public async Task<WhoamiResponse> WhoamiAsync(CancellationToken ct = default)
    {
        var node = await _transport.RequestAsync(RequestOptions.Get("/whoami"), ct).ConfigureAwait(false);
        return Json.Deserialize<WhoamiResponse>(node);
    }
}
